In [37]:
# Importação das bibliotecas utilizadas na atividade

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [38]:
df = pd.read_csv('data/heart.csv')

In [39]:
df.shape

(1025, 14)

In [40]:
df.head()

,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
0,52,1,0,125,212,0,1,168,0,1.0,2,2,3,0
1,53,1,0,140,203,1,0,155,1,3.1,0,0,3,0
2,70,1,0,145,174,0,1,125,1,2.6,0,0,3,0
3,61,1,0,148,203,0,1,161,0,0.0,2,1,3,0
4,62,0,0,138,294,1,1,106,0,1.9,1,3,2,0


## Informações de atributos do dataset

- **age** → idade da pessoa | Numérico Contínuo
- **sex** → sexo *(1 = homem, 0 = mulher)* | Categórico Binário
- **cp** → tipo de dor no peito *(4 categorias)* | Categórico Nominal
- **trestbps** → pressão arterial em repouso *(mmHg)* | Numérico Contínuo
- **chol** → nível de colesterol *(mg/dl)* | Numérico Contínuo
- **fbs** → glicemia alta em jejum *(1 = sim, 0 = não)* | Categórico Binário 
- **restecg** → resultado do eletrocardiograma *(0, 1, 2)* | Categórico Nominal
- **thalach** → frequência cardíaca máxima atingida | Numérico Contínuo
- **exang** → angina induzida por exercício *(1 = sim, 0 = não)* | Categórico Binário
- **oldpeak** → alteração cardíaca causada por esforço | Numérico Contínuo
- **slope** → inclinação dessa alteração durante exercício | Categórico Ordinal
- **ca** → número de vasos sanguíneos afetados *(0–3)* | Numérico Discreto
- **thal** → tipo de problema cardíaco *(0 = normal, 1 = fixo, 2 = reversível)* | Categórico Nominal

# Tratamento de Dados

In [41]:
# Identificar a quantidade de nulos por coluna

print(df.isnull().sum())

age         0
sex         0
cp          0
trestbps    0
chol        0
fbs         0
restecg     0
thalach     0
exang       0
oldpeak     0
slope       0
ca          0
thal        0
target      0
dtype: int64


In [42]:
# Identificar a quantidade de duplicatas por coluna

df.duplicated().sum()

np.int64(723)

In [43]:
df[df.duplicated()]

,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
15,34,0,1,118,210,0,1,192,0,0.7,2,0,2,1
31,50,0,1,120,244,0,1,162,0,1.1,2,0,2,1
43,46,1,0,120,249,0,0,144,0,0.8,2,0,3,0
55,55,1,0,140,217,0,1,111,1,5.6,0,0,3,0
61,66,0,2,146,278,0,0,152,0,0.0,1,1,2,1
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1020,59,1,1,140,221,0,1,164,1,0.0,2,0,2,1
1021,60,1,0,125,258,0,0,141,1,2.8,1,1,3,0
1022,47,1,0,110,275,0,0,118,1,1.0,1,1,2,0
1023,50,0,0,110,254,0,0,159,0,0.0,2,0,2,1


## 🚨🚨 IMPORTANTE 🚨🚨

Para fins educacionais, o criador do dataset original 'inflou' o conjunto de dados colocando elementos repetidos! De 1025 elementos no dataset, 723 são repetidos.
Portanto, devemos eliminar esses elementos duplicados

In [44]:
df_clean = df.drop_duplicates()
df_clean

,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
0,52,1,0,125,212,0,1,168,0,1.0,2,2,3,0
1,53,1,0,140,203,1,0,155,1,3.1,0,0,3,0
2,70,1,0,145,174,0,1,125,1,2.6,0,0,3,0
3,61,1,0,148,203,0,1,161,0,0.0,2,1,3,0
4,62,0,0,138,294,1,1,106,0,1.9,1,3,2,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
723,68,0,2,120,211,0,0,115,0,1.5,1,0,2,1
733,44,0,2,108,141,0,1,175,0,0.6,1,0,2,1
739,52,1,0,128,255,0,1,161,1,0.0,2,1,3,0
843,59,1,3,160,273,0,0,125,0,0.0,2,0,2,0
